maven-core/src/main/java/org/apache/maven/project/collector/MultiModuleCollectionStrategy.java

```diff
@@ -64,23 +65,36 @@ public MultiModuleCollectionStrategy( ModelLocator modelLocator, ProjectsSelecto
     @Override
     public List<MavenProject> collectProjects( MavenExecutionRequest request ) throws ProjectBuildingException
     {
-        File moduleProjectPomFile = getMultiModuleProjectPomFile( request );
-        List<File> files = Collections.singletonList( moduleProjectPomFile.getAbsoluteFile() );
+        Optional<File> moduleProjectPomFile = getMultiModuleProjectPomFile( request );
+        File absoluteFile = moduleProjectPomFile.orElse( request.getPom() ).getAbsoluteFile();
+        List<File> files = Collections.singletonList( absoluteFile );
         try
         {
             List<MavenProject> projects = projectsSelector.selectProjects( files, request );
+            if ( moduleProjectPomFile.isPresent()
+                    && projects.stream()
```

Comment:
For better understanding of the `if` statement, I think it would help if the second part would be refactored to a separate boolean variable with a meaningful name.